### ISSUE TITLE
Rule proposal: removing unnecessary checks around _.isEmpty

### ISSUE THREAD

**Issue by mik01aj** [wix/eslint-plugin-lodash#83]
in my codebase, I often see a pattern like this:

    if (!campaigns || _.isEmpty(campaigns)) {

but it is equivalent to:

    if (_.isEmpty(campaigns)) {

Could we have a rule to detect cases like this?

**Comment by mik01aj** [comment]
Also, this is pretty similar: `_.isObject(x) && _.isArray(x)` is equivalent to just `_.isArray(x)`.

**Comment by ganimomer** [comment]
I think these might be two different rules.
And I see these patterns around as well, but what would you suggest warning against?
I can think of these patterns:
```js
(!x || _.isEmpty(x))
(x === undefined || _.isEmpty(x))
(x === null || _.isEmpty(x))
```
Can you think of more examples? Do you think this should be in the same rule as the _.isObject case? Something like `no-unnecessary-checks`?

**Comment by mik01aj** [comment]
I just tried to find some more examples, and realized that `!x || _.isEmpty(x)` is *not always* equivalent to just `_.isEmpty(x)`, for example when x=1. (`!x` is `false` and `_.isEmpty(x)` is `true`).
